src/git.ts

```diff
@@ -0,0 +1,1264 @@
+                default:
+                    invalidCommitOptionCheck = true
+                    //invalidCommitOptions.
+            }
+        })
+        if (invalidCommitOptionCheck === true) {
+            NEURO.client?.sendActionResult(actionData.id, false, `Invalid commit options: ${invalidCommitOptions.map((option: string) => {option})}`)
+            return;
+        }
+    }
+
+    NEURO.client?.sendActionResult(actionData.id, true)
+
+    repo.inputBox.value = message;
+    repo.commit(message).then(() => {
+        NEURO.client?.sendContext(`Committed with message: "${message}"\nCommit options used: ${commitOptions ? commitOptions : "None"}`);
+    }, (err: string) => {
+        NEURO.client?.sendContext(`Failed to record commit`);
+        logOutput("ERROR", `Failed to commit: ${err}`)
+    });
+}
+
+export function handleGitMerge(actionData: any) {
+    if (!git) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'Git extension not available.');
+        return;
+    }
+    if (!vscode.workspace.getConfiguration('neuropilot').get('permission.gitOperations', false)) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You do not have permission to perform Git operations.');
+        return;
+    }
+    if (!repo) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You are not in a git repository.');
+        return;
+    }
+
+    const refToMerge = actionData.params?.ref_to_merge
+
+    if (!repo.getBranch(refToMerge)) {
+        NEURO.client?.sendActionResult(actionData.id, false, "This branch does not exist. Perhaps you misspelled it?")
+        return;
+    }
+
+    NEURO.client?.sendActionResult(actionData.id, true)
+
+    repo.merge(refToMerge).then(() => {
+        NEURO.client?.sendContext(`Cleanly merged ${refToMerge} into the current branch.`)
+    }, (err: string) => {
+        NEURO.client?.registerActions([
+            {
+                name: 'abort_merge',
+                description: 'Aborts the current merge oepration.',
+                schema: {}
+            },
+        ])
+        NEURO.client?.sendContext(`Couldn't merge ${refToMerge}: ${err}`)
+        logOutput("ERROR", `Encountered an error when merging ${refToMerge}: ${err}`)
+    })
+}
+
+export function handleAbortMerge(actionData: any) {
+    if (!git) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'Git extension not available.');
+        return;
+    }
+    if (!vscode.workspace.getConfiguration('neuropilot').get('permission.gitOperations', false)) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You do not have permission to perform Git operations.');
+        return;
+    }
+    if (!repo) {
+        NEURO.client?.sendActionResult(actionData.id, true, 'You are not in a git repository.');
+        return;
+    }
+
+    NEURO.client?.sendActionResult(actionData.id, true)
+
+    repo.mergeAbort().then(() => {
+        NEURO.client?.unregisterActions(["abort_git_merge"])
```

Comment:
Incorrect action name
```suggestion
        NEURO.client?.unregisterActions(["abort_merge"])
```